### DISCUSSION TITLE
Ideas for Future Updates: (Collection)

### DISCUSSION THREAD

**Discussion by Unlucky-Life** [Unlucky-Life/ankimon#62] [Announcements]
- [ ] - Battle against Trainers aswell in reviewer
- [ ] - AnkimonShop inside of Anki to buy items
- [ ] - TM/VW use
- [ ] - Arena Battles and Elite 4 Battles
- [ ] - switching out pokemon in battle
- [ ] - Using diffrent items in battle
- [ ] - allowing to choose movesets with popup screen
- [ ] - Maybe being able to use pokeballs to catch pokemon aswell
- [ ]  - Receive and Hit with damaged based on answer quality and speed
- [ ]  - Time inside of reviewer
- [ ]  - being able to also have you pokemon be ko'd and then heal in pokemon center
- [ ] - be able to train pokemon beyond level 100
- [ ] - be able to always train pokemon past their evolevel
- [ ] - be able to give pokemo everstone to keep prevo form
- [ ]  - more pokemon forms - mega, etc
- [ ] - Having 5 and 6 as shortcuts for defeating and catching pokemon
- [ ] - being able to switch out mainpokemon mid fight
- [ ] - being able to remove complete ankimon ui in reviewer
- [ ] - Trainercard that levels by gaining xp through training pokemon, amount of badges, cards studies, streaks, amount of pokemon caught
- [x] - friendship level with certain pokemon
- [ ] - date of caught pokemon
- [x] - more filter options for pokedex such as: type, date caught, shiny, etc.
- [ ] - be able to switch out mainpokemon during battle
- [ ] - be able to choose region => based on region battle only pokemons in that region
- [ ] - probability to capture rare pokemons increases per day for those reviews - not per session
- [ ] - more xp on defeating a pokemon based on rarity
- [ ] - favorite 6 Pokemon
- [ ] - reread popup messages
- [ ]

**Comment by inww7** [comment]
我能提一个小的想法吗, 我提议**能否添加一个选项, 彻底隐藏主界面的元素** , 考试迫在眉睫, 我很难再把我的注意力分给那些可爱的精灵, 但我又可惜如果关掉插件, 那么我的"固拉多"就不能继续升级了。而我尝试过将修改设置, 但无法做到复习页面全部元素隐藏, 最终只能妥协的将左下角小小的宝可梦以及它的血条保留下来。
我想这个需求可能不少人都会存在吧~ 能将它安排到下一个版本吗。

等我高中毕业之后可能会尝试捐赠这个插件计划, 在我看来这真是个具有野心和前景的插件。

**Reply by Unlucky-Life** [reply]
for my non chinese speakers here is the translation of the comment above: 
"May I suggest a small idea, I propose to add an option to completely hide the elements of the main interface, the exam is imminent, I can hardly give my attention to those cute sprites, but I regret that if I turn off the plugin, then I can not continue upgrading my “Gulador”. I tried to change the settings, but I couldn't hide all the elements on the review page, so I compromised by keeping the little Pokémon in the bottom left corner and its blood bar.
I think this is a request that a lot of people have~ Can you schedule it for the next version.

I might try to donate to the plugin program after I graduate from high school, it seems like a really ambitious and promising plugin to me.

Translated with DeepL.com (free version)"

**Reply by Unlucky-Life** [reply]
@goodbird2 
所以，如果我理解正确的话 
您是否要求在审稿人中提供一个隐藏所有鬼门元素的选项？
(我认为这可以在下次更新时轻松实现）。
现在您可以
- 关闭/隐藏：
-- 弹出对话框
-- 宠物小精灵的hp条（相反，宠物小精灵的hp会以文本形式显示）
-- 主小精灵的精灵（图像
-- 主小精灵的 xp 栏
-- 宠物小精灵的动画和声音

你还想隐藏什么？
(我对此持非常开放的态度，但我只需要知道哪些需要整合，哪些不需要整合，以及如何实现这些功能）。
比如说，我们可以基本上隐藏审查员中的整个宠物小精灵元素，或者在选项中删除每个元素。

**Reply by inww7** [reply]
@Unlucky-Life 
是的, 正如您最后一段所言, 我认为应当提供一个选项, 来选择是否隐藏审查(review)中的**所有**宝可梦的图片和血条, 包括文本形式的血条等等。

**Reply by Unlucky-Life** [reply]
@goodbird2
应该很容易整合！
意思是
我们将摆脱看到自己的小精灵以及野生小精灵和 hp 显示及名称等的困扰。

这应该是可行的。
然后，这些用户将需要使用宠物小精灵窗口来获取更多信息，并捕捉或训练你的宠物小精灵。

**Reply by Unlucky-Life** [reply]
@goodbird2 Check this out: https://github.com/Unlucky-Life/ankimon/releases/download/1.28/ankimon-1.28-anki21-ankiweb.ankiaddon Here you can set "styling in reviewer" to false and wont have any ui components in reviewer :)

请查看： https://github.com/Unlucky-Life/ankimon/releases/download/1.28/ankimon-1.28-anki21-ankiweb.ankiaddon 在这里，您可以将 “审阅器中的样式 ”设置为 false，这样审阅器中就不会有任何 UI 组件了：)

**Reply by inww7** [reply]
Dear @Unlucky-Life , this is the latest errors for V1.28-yesterday: 

Anki 24.04.1 (ccd9ca1a)  (ao)
Python 3.9.18 Qt 6.6.2 PyQt 6.6.1
Platform: Windows-10-10.0.19041

When loading Ankimon:
Traceback (most recent call last):
  File "C:\Users\*\AppData\Roaming\Anki2\addons21\1908235722\__init__.py", line 1134, in generate_random_pokemon
    base_experience = search_pokeapi_db_by_id(id, "base_experience")
  File "C:\Users\*\AppData\Roaming\Anki2\addons21\1908235722\__init__.py", line 1995, in search_pokeapi_db_by_id
    with open(str(pokeapi_db_path), "r", encoding="utf-8") as json_file:
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\*\\AppData\\Roaming\\Anki2\\addons21\\1908235722\\user_files\\data_files\\pokeapi_db.json'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "aqt.addons", line 247, in loadAddons
  File "C:\Users\*\AppData\Roaming\Anki2\addons21\1908235722\__init__.py", line 2052, in <module>
    name, id, level, ability, type, stats, enemy_attacks, base_experience, growth_rate, hp, max_hp, ev, iv, gender, battle_status, battle_stats = generate_random_pokemon()
  File "C:\Users\*\AppData\Roaming\Anki2\addons21\1908235722\__init__.py", line 1168, in generate_random_pokemon
    showInfo("Error", "Can't open the JSON File.")
  File "aqt.utils", line 296, in showInfo
TypeError: arguments did not match any overloaded call:
  QMessageBox(parent: Optional[QWidget] = None): argument 1 has unexpected type 'str'
  QMessageBox(icon: QMessageBox.Icon, title: Optional[str], text: Optional[str], buttons: QMessageBox.StandardButton = QMessageBox.NoButton, parent: Optional[QWidget] = None, flags: Qt.WindowType = Qt.Dialog|Qt.MSWindowsFixedSizeDialogHint): argument 1 has unexpected type 'str'

**Reply by Unlucky-Life** [reply]
@goodbird2  Hmmm ok let me see 

Aswell try installing the newest prerelease: https://github.com/Unlucky-Life/ankimon/releases/tag/1.281

so it seems like you are missing some files like pokeapi_db.json file. You can get that from here:

Make sure to backup your `mypokemon.json`, `items.json`, `badges.json` and `mainpokemon.json` files before !

https://github.com/Unlucky-Life/ankimon/issues/32
=> You need to place these files into your user_files folder

**Reply by inww7** [reply]
@Unlucky-Life  Thx!! My problem solved! I love this version!
